From: Andrea Azzarone <andrea.azzarone@canonical.com>
Date: Tue, 8 May 2018 17:16:03 +1200
Subject: Ensure stamp file if the user quit the wizard

---
 gnome-initial-setup/gis-driver.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gnome-initial-setup/gis-driver.c b/gnome-initial-setup/gis-driver.c
index daf3a43..77434ee 100644
--- a/gnome-initial-setup/gis-driver.c
+++ b/gnome-initial-setup/gis-driver.c
@@ -589,6 +589,17 @@ gis_driver_is_small_screen (GisDriver *driver)
   return driver->small_screen;
 }
 
+static void
+gis_driver_shutdown (GApplication *app)
+{
+  GisDriver *driver = GIS_DRIVER (app);
+
+  G_APPLICATION_CLASS (gis_driver_parent_class)->shutdown (app);
+
+  if (driver->mode == GIS_DRIVER_MODE_EXISTING_USER)
+    gis_ensure_stamp_files (driver);
+}
+
 static gboolean
 monitor_is_small (GdkMonitor *monitor)
 {
@@ -873,6 +884,7 @@ gis_driver_class_init (GisDriverClass *klass)
   gobject_class->finalize = gis_driver_finalize;
   application_class->startup = gis_driver_startup;
   application_class->activate = gis_driver_activate;
+  application_class->shutdown = gis_driver_shutdown;
 
   signals[REBUILD_PAGES] =
     g_signal_new ("rebuild-pages",
